popde/utils_plot.py

```diff
@@ -0,0 +1,164 @@
+                             np.max(self.kde.data[:, dim_idx]),
+                             num_points)
+        points = np.zeros((num_points, self.kde.data.shape[1]))
+        points[:, dim_idx] = values
+        density_values = self.kde.evaluate(points)
+
+        plt.plot(values, density_values, **kwargs)
+        plt.xlabel(self.kde.dim_names[dim_idx] if self.kde.dim_names else f'Dimension {dim_idx + 1}')
+        plt.ylabel('Density')
+        plt.title(f'1D Projection along {self.kde.dim_names[dim_idx]}'
+                  if self.kde.dim_names else f'1D Projection along Dimension {dim_idx + 1}')
+        plt.show()
+
+    def plot_2d_slice(self, dim1_idx, dim2_idx, num_points=100, **kwargs):
+        """
+        Plot a 2D slice of the KDE by fixing two dimensions.
+
+        Parameters:
+        -----------
+        dim1_idx : int
+            The index of the first dimension.
+        dim2_idx : int
+            The index of the second dimension.
+        num_points : int, optional
+            The number of points to evaluate the slice, default is 100.
+        **kwargs:
+            Additional keyword arguments to pass to the plot function.
+        """
+        values_dim1 = np.linspace(np.min(self.kde.data[:, dim1_idx]),
+                                  np.max(self.kde.data[:, dim1_idx]),
+                                  num_points)
+        values_dim2 = np.linspace(np.min(self.kde.data[:, dim2_idx]),
+                                  np.max(self.kde.data[:, dim2_idx]),
+                                  num_points)
+        points = np.zeros((num_points ** 2, self.kde.data.shape[1]))
+        points[:, dim1_idx] = np.tile(values_dim1, num_points)
+        points[:, dim2_idx] = np.repeat(values_dim2, num_points)
+        density_values = self.kde.evaluate(points)
+
+        plt.tricontourf(points[:, dim1_idx], points[:, dim2_idx], density_values, **kwargs)
+        plt.xlabel(self.kde.dim_names[dim1_idx] if self.kde.dim_names else f'Dimension {dim1_idx + 1}')
+        plt.ylabel(self.kde.dim_names[dim2_idx] if self.kde.dim_names else f'Dimension {dim2_idx + 1}')
+        plt.title(f'2D Slice along {self.kde.dim_names[dim1_idx]} and {self.kde.dim_names[dim2_idx]}'
+                  if self.kde.dim_names else f'2D Slice along Dimensions {dim1_idx + 1} and {dim2_idx + 1}')
+        plt.show()
+
+# Example usage:
+from density_estimate  import  SimpleKernelDensityEstimation
+mean1, sigma1 = 14.0, 1.5
+mean2, sigma2 = 3.0, 0.25
+n_samples = 1000
+rndgen = np.random.RandomState(seed=1)
+sample1 = rndgen.normal(mean1, sigma1, size=n_samples)
+sample2 = rndgen.normal(mean2, sigma2, size=n_samples)
+data= np.column_stack((sample1, sample2)) # shape is (n_points, n_features)
+kde_instance = SimpleKernelDensityEstimation(data, dim_names=['m1', 'z'])
+kde_plotter = KDEPlotter(kde_instance)
+kde_plotter.plot_1d_projection(0)
+kde_plotter.plot_2d_slice(0, 1)
+
+
+class DataVisualizer:
```

Comment:
I don't know what this class should do. The point of creating a class is to organize some information that belongs together and should persist through several operations.

Reply:
I removed this.